add a new dotnet11 plugin and a new system text json skill to it. (#535)
dotnet/skills: plugins/dotnet11/skills/system-text-json-net11/SKILL.md

---
name: system-text-json-net11
description: >

>   Provides guidance on new System.Text.Json APIs introduced in .NET 11.

Reviewer: What's the purpose of a skill that enumerates new features in a particular release of a library? If it's to upgrade specific coding patterns to take advantage of new APIs, these coding patterns should be enumerated explicitly, e.g. with *BEFORE* and *AFTER* sections.

Author: the purpose is mostly to introduce new features to the model (something that model was not trained on), not necessarily to upgrade.

Reviewer: Right, in that case wouldn't it make sense to exactly document what the new features are meant to solve? I don't see how simply enumerating new APIs will add much to the context.

Author: There have been several experiments that have shown that just listing out the new APIs and some examples on how to use them leads to significantly better results. Our own results and experiments with NuGet team shows that. Here is a blog post from google on this - https://developers.googleblog.com/en/closing-the-knowledge-gap-with-agent-skills/

  It covers typed JsonTypeInfo access via GetTypeInfo<T> and TryGetTypeInfo<T> on
  JsonSerializerOptions, and the new JsonNamingPolicy.PascalCase static property.
  Use when serializing or deserializing JSON in .NET 11 applications and needing
  typed metadata access or PascalCase property naming.
---

# System.Text.Json — .NET 11

> New APIs added to `System.Text.Json` across .NET 11 releases.

Reviewer: Do note that there are whole load of STJ features still in the pipeline, so this document would need to be kept in sync.

Author: Yes, it is expected that we will update these before final release of .NET 11

## Target Framework

```xml
<TargetFramework>net11.0</TargetFramework>
```

> ## New APIs

Author: Fixed

### Typed `JsonTypeInfo` Access

#### `JsonSerializerOptions.GetTypeInfo<T>()`

Returns a strongly-typed `JsonTypeInfo<T>` for the specified type, using the
options' configured type-info resolver.

```csharp

> JsonTypeInfo<T> GetTypeInfo<T>()

Reviewer: Why not inline examples into each section? This alone doesn't seem to provide a whole lot of context on its own?

Author: I played with inlining the examples as well as putting them in the end and the results were similar. I had decided to put it in the end so that, where needed, we can cover multiple APIs in the same example (to reduce total text).

---
name: system-text-json-net11
description: >
  Provides guidance on new System.Text.Json APIs introduced in .NET 11.

>   It covers typed JsonTypeInfo access via GetTypeInfo<T> and TryGetTypeInfo<T> on

Author: fixed

  JsonSerializerOptions, and the new JsonNamingPolicy.PascalCase static property.
  Use when serializing or deserializing JSON in .NET 11 applications and needing
  typed metadata access or PascalCase property naming.
---

# System.Text.Json — .NET 11

New APIs added to `System.Text.Json` across .NET 11 releases.

## When to Use

- Serializing or deserializing JSON in a .NET 11 (or later) project
- Needing strongly-typed `JsonTypeInfo<T>` access instead of the untyped `JsonTypeInfo` overload
- Wanting to safely check whether type metadata is available without catching exceptions (`TryGetTypeInfo<T>`)
- Requiring PascalCase property naming during JSON serialization

## When Not to Use

- The project targets .NET 10 or earlier — these APIs are not available before .NET 11

> - Using a third-party JSON library (e.g., Newtonsoft.Json) instead of `System.Text.Json`

Reviewer 2: Consider: I don't know if the model understands third-party in this context.

```suggestion
- Using a JSON library that is not `System.Text.Json` (e.g., Newtonsoft.Json)
```

Author: fixed
